File: src/main/java/moa/moabackend/global/error/ErrorResponse.java
package moa.moabackend.global.error;

import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class ErrorResponse {

    private String code;
    private String message;

    public static ErrorResponse of(String code, String message) {
        return ErrorResponse.builder()
                .code(code)
                .message(message)
                .build();
    }
}

File: src/main/kotlin/moa/moabackend/global/error/ErrorResponse.kt
package moa.moabackend.global.error

import moa.moabackend.global.error.exception.ErrorCode

class ErrorResponse(
    val status: Int,
    val message: String

) {
    companion object {
        fun of(errorCode: ErrorCode, message: String): ErrorResponse {
            return ErrorResponse(
                status = errorCode.status,
                message = message
            )
        }
    }
}
